lib/index.ts

```diff
@@ -463,6 +473,25 @@ export class DockerProgress {
 		return hash;
 	}
 
+	/** Build docker image calling onProgress with extended progress info regularly */
+	async build(
+		tarStream: Stream.Readable,
+		onProgress: ProgressCallback,
+		options?: BuildOptions,
+	): Promise<string> {
+		const ignoreErrorEvents = !!options?.ignoreProgressErrorEvents;
+		const reporter = await this.getProgressReporter();
+		onProgress = reporter.pullProgress(onProgress);
+		const stream = await this.docker.buildImage(tarStream, options);
+		const hash = await awaitRegistryStream(
+			stream,
+			onProgress,
+			ignoreErrorEvents,
+		);
+
+		return hash;
```

Comment:
Looking good. 👍  Were you able to confirm that this `hash` value is correctly determined for the built image? As in a [previous comment](https://github.com/balena-io-modules/docker-progress/pull/55#discussion_r756432309) -
> Just wondering if the progress output for pull, push and build are really sufficiently compatible, such that `awaitRegistryStream` and `tryExtractDigestHash`, which were implemented for `push` and `pull`, work equally well for `build`.

Reply:
I haven't gotten to that yet, but I will. Right now I'm trying to figure out why the `dockerode.buildImage` doesn't seem to use the authentication options in `authconfig`

Reply:
This seems relevant to my issue https://github.com/apocas/dockerode/issues/618

Reply:
Ok, it looks like the `/build` docker engine endpoint does not support the `X-Registry-Auth` header and requires the `X-Registry-Config` header with the following format.

`{'https://index.docker.io/v1/': {username: 'myusername', password: '<mypass>'}}`

Which make sense, since a build may require to authenticate to multiple registries, but it makes the `BuildOptions` interface wrong.

Reply:
I'll test this tomorrow on the supervisor and check the returning hash @pdcastro